internal/k6runner/k6runner.go

```diff
@@ -33,6 +41,31 @@ type Settings struct {
 	Timeout int64 `json:"timeout"`
 }
 
+// CheckInfo holds information about the SM check that triggered this script.
+type CheckInfo struct {
+	// Type is the string representation of the check type this script belongs to (browser, scripted, multihttp, etc.)
+	Type string `json:"type"`
+	// Metadata is a collection of key/value pairs containing information about this check, such as check and tenant ID.
+	// It is loosely typed on purpose: Metadata should only be used for informational properties that will make its way
+	// into telemetry, and not for making decision on it.
+	Metadata map[string]any `json:"metadata"`
+}
+
+// FromSM populates the given Check from the information of the SM check. Existing fields are overwritten.
+func (c CheckInfo) FromSM(smc sm.Check) CheckInfo {
+	c.Type = smc.Type().String()
+
+	if c.Metadata == nil {
+		c.Metadata = map[string]any{}
+	}
+	c.Metadata["id"] = smc.Id
+	c.Metadata["tenantID"] = smc.TenantId
+	c.Metadata["created"] = smc.Created
+	c.Metadata["modified"] = smc.Modified
```

Comment:
I don't love the merge nature of this, and I do have a problem with the signature of the method.

```
anotherCheckInfoWithRandomGarbageInIt := checkInfoWithRandomGarbageInIt.FromSM(check)
```

`anotherCheckInfoWithRandomGarbageInIt` has the same map as `checkInfoWithRandomGarbageInIt`. That's not cool. Please note: https://go.dev/play/p/B9HNY11N8c4

This code in the test is weird:

```
ci := CheckInfo{}.FromSM(check)
```

I would prefer a simple constructor approach:

```
func NewCheckInfo(check sm.Check) CheckInfo

ci := NewCheckInfo(check)
```

If you need the merging semantics, I would need to see an example to understand why.

Reply:
Mostly out of curiosity, why do you want created and modified in there?

If anything, I would add the version...

Reply:
And if you are including the tenant ID you *must* include the region ID.

Reply:
> And if you are including the tenant ID you must include the region ID.
That's a good catch, I definitely missed that

> Mostly out of curiosity, why do you want created and modified in there?
No particular reason other than "seemed vaguely useful for debugging". Might be useful if we want to plot the age distribution from logs? Again, no strong reason.

Reply:
Changed the weird merge thingie to a constructor.

Reply:
> That's a good catch, I definitely missed that. ... Where is it tho? I cannot find it in the check.

Argh. You have the wrong type of check because you are in the Prober, which receives an sm.Check, and you want a model.Check.

in scraper.go, in `collectData` you still have access to it because the scraper holds the correct type of check.

This calls `getProbeMetrics`, which no longer has access to that information.

This calls `runProber`, which is what calls `prober.Probe`, which is what ends up calling

Reply:
That's another of those 100 line changes across internal tests and tens of implementations, unfortunately. It will take some time.